controllers/securitypolicy_controller.go

```diff
@@ -140,33 +140,48 @@ func (r *SecurityPolicyReconciler) spec(
 // updateAccess updates given AccessDefinition's APIID and APIName fields based on ApiDefinition CR that is referred
 // in the AccessDefinition. So that, it includes k8s details of the referred ApiDefinitions.
 func (r *SecurityPolicyReconciler) updateAccess(ctx context.Context, ad *tykv1.AccessDefinition) error {
+	var apiID, apiName string
 	api := &tykv1.ApiDefinition{}
 	if err := r.Get(ctx, types.NamespacedName{Name: ad.Name, Namespace: ad.Namespace}, api); err != nil {
 		if errors.IsNotFound(err) {
-			r.Log.Info("ApiDefinition resource not found. Unable to attach to SecurityPolicy. ReQueue",
+			r.Log.Info("ApiDefinition resource not found. Unable to attach to SecurityPolicy. looking for OasApiDefinition",
 				"Name", ad.Name,
 				"Namespace", ad.Namespace,
 			)
-
-			return err
 		}
+	}
 
-		r.Log.Error(err, "Failed to get APIDefinition to attach to SecurityPolicy")
+	apiID = api.Status.ApiID
+	apiName = api.Spec.Name
 
-		return err
+	if apiID == "" {
+		r.Log.Info("Failed to get normal APIDefinition to attach to SecurityPolicy, looking for OasApiDefiniiton")
```

Comment:
the same log is duplicated.

Reply:
The log is not duplicated as it refers to a different case than the log above.